devops/setup-gpg/action.yaml

```diff
@@ -1,52 +1,72 @@
 name: 'Setup GPG'
 description: 'Configures this action to run gpg with a given key and pass'
 inputs:
-  gpg-private-key:  # id of input
+  gpg-private-key:
+    description: 'GPG private key exported as an ASCII armored version or its base64 encoding'
     required: true
-  gpg-key-pass:  # id of input
+  gpg-key-pass:
+    description: 'Passphrase of the GPG private key'
     required: true
-  gpg-key-name:  # id of input
+  gpg-public-key:
+    description: 'GPG public key exported as an ASCII armored version or its base64 encoding'
     required: true
-    default: "Aerospike"
+  gpg-key-name:
+    description: 'GPG key name'
+    required: true
+    default: "aerospike-inc"
+  gpg-trust-level:
+    description: 'Set key trust level'
+    required: false
+    default: 5
 runs:
   using: "composite"
   steps:
-      - name: "check if private key is not empty"
+      - name: "check if private key is empty"
         env: 
-            PRIVATE_KEY: ${{ inputs.gpg-private-key }}
-        if: ${{ env.PRIVATE_KEY == '' }}
+            GPG_PRIVATE_KEY: ${{ inputs.gpg-private-key }}
+        if: ${{ env.GPG_PRIVATE_KEY == '' }}
+        run: |
+          echo "The gpg-private-key was empty"
+          exit 1
+        shell: bash
+      - name: "check if public key is empty"
+        env:
+            GPG_PUBLIC_KEY: ${{ inputs.gpg-public-key }}
+        if: ${{ env.GPG_PUBLIC_KEY == '' }}
         run: |
-          echo "the gpg-private-key was empty"
+          echo "The gpg-public-key was empty"
           exit 1
         shell: bash
-      - name: "check if key name is not empty"
+      - name: "check if key name is empty"
         env: 
-            KEY_NAME: ${{ inputs.gpg-key-name }}
-        if: ${{ env.KEY_NAME == '' }}
+            GPG_ID: ${{ inputs.gpg-key-name }}
+        if: ${{ env.GPG_ID == '' }}
         run: |
-          echo "the gpg-key-name was empty"
+          echo "The gpg-key-name was empty"
           exit 1
         shell: bash
-      - name: "check if key pass is not empty"
+      - name: "check if key pass is empty"
         env: 
-            KEY_PASS: ${{ inputs.gpg-key-pass }}
-        if: ${{ env.KEY_PASS == '' }}
+            GPG_KEY_PASS: ${{ inputs.gpg-key-pass }}
+        if: ${{ env.GPG_KEY_PASS == '' }}
         run: |
-          echo "the secret gpg-key-pass was empty"
+          echo "The secret gpg-key-pass was empty"
           exit 1
         shell: bash
       - name: install tools
         run: |
-          sudo apt-get update && sudo apt-get install ca-certificates gnupg dpkg-dev -y
+          sudo apt-get update && sudo apt-get install ca-certificates gnupg dpkg-dev dpkg-sig rpm -y
```

Comment:
is this needed? In all my testing rpm and dpkg tools where installed by default on the workers. I'm not sure if we should do this here anyhow... this is about setting up gpg, not installing any and all tools. I will think about this (though i'm still not sure it's needed at all?)

Reply:
Not sure if dpkg-sig available by default on ubuntu:latest.  If you know it's then we can remove dpkg-sig.  It on't hurt if we keep it there just in case.

Reply:
chatgpt says both rpm & dpkg-sig not available in ubuntu latest (24.04), so looks like we need them installed.  May be github action has them installed as default?  I recalled my action failed because they not available.  I'll test again to confirm.

Reply:
Action failed after removing rpm & dpkg-sig from installation.
/home/runner/work/_temp/4d5835d7-c72c-46c8-b3b0-2bccfc32e931.sh: line 1: dpkg-sig: command not found

The rpm still works.  My recommendation is keep them both.

Reply:
yeah. chatgpt can't be trusted :D at all. I'm not 100% sold on installing things but joe probably would like it. so this is fine. Thanks for investigating!!

Reply:
I went here also after reading through these comments:
https://launchpad.net/ubuntu/noble/+localpackagediffs?field.name_filter=dpkg-sig&field.package_type=all&field.package_type-empty-marker=1

rpm: present
dpkg-sig: not present